File: flask-backend/controllers/page_rank_controller.py
import os
import requests
from urllib.parse import urlparse

def get_page_rank(url):
    try:
        domain = urlparse(url).netloc
        api_key = os.getenv('OPENPAGE_API_KEY')
        
        if not api_key:
            return {'page_rank': 0}

        response = requests.get(
            f'https://openpagerank.com/api/v1.0/getPageRank?domains[]={domain}',
            headers={'API-OPR': api_key}
        )
        
        data = response.json()
        
        if data.get('status_code') == 200:
            return {'page_rank': data['response'][0]['page_rank_decimal']}
        
        return {'page_rank': 0}
    except Exception as e:
        print(f'PageRank Error: {str(e)}')
        return {'page_rank': 0}

File: nodejs-backend/pageRankController.js
async function getPageRank(url) {
    try {
        const domain = new URL(url).hostname;
        
        const response = await fetch(
            `https://openpagerank.com/api/v1.0/getPageRank?domains[]=${domain}`,
            {
                headers: {
                    'API-OPR': process.env.OPENPAGE_API_KEY
                }
            }
        );
        
        const data = await response.json();
        
        if (data.status_code === 200 && data.response[0].status_code === 200) {
            return { pageRank: data.response[0].page_rank_decimal };
        } else {
            return { pageRank: 0 };
        }
    } catch (error) {
        console.error('PageRank Error:', error);
        return { pageRank: 0 };
    }
};

export default getPageRank;